riot/riot.py

```diff
@@ -557,15 +598,24 @@ def prepare(
 
             pkg_str = self.pkg_str
             assert pkg_str is not None
+            compiled_requirements_file = (
+                f"{DEFAULT_RIOT_PATH}/requirements/{self.short_hash}.txt"
+            )
+            if recompile_reqs or not os.path.exists(compiled_requirements_file):
+                _ = self.requirements
+            cmd = (
+                f"pip --disable-pip-version-check install --prefix '{self.prefix}' --no-warn-script-location "
+                f"-r {compiled_requirements_file}"
+            )
```

Comment:
I am not sure I follow this correctly.

We always require the requirements file?

Reply:
That's right. The .txt lockfile is always part of the `run` command. If it doesn't exist it's created, but if it does it's used and the work of creating it is skipped.

Reply:
oh! the `_ = self.requirements` is the part that threw me off

I wonder if we should not have that as a property but a function to be more explicit about it. wdyt?